Makefile

```diff
@@ -61,6 +63,21 @@ install_system_db_dependencies: apt_get_update ## Install system-level DB depend
 
 install_system_dependencies: apt_get_update ## Install system-level dependencies from `debian_packages.lst`. Ignores comments.
 	sudo -E apt-get install -y `grep -v '^#' debian_packages.lst | tr -d '\r'`
+	if [ -z $$CI ] ; then \
+		echo "Installing Firefox because we're not in a CI."; \
+		sudo apt-get install -y libgtk3.0-cil-dev libasound2 libasound2 libdbus-glib-1-2 libdbus-1-3; \
+		sudo curl -sL -o /tmp/firefox.deb 'https://s3.amazonaws.com/circle-downloads/firefox-mozilla-build_47.0.1-0ubuntu1_amd64.deb'; \
+		echo 'ef016febe5ec4eaf7d455a34579834bcde7703cb0818c80044f4d148df8473bb  /tmp/firefox.deb' | sha256sum -c; \
+		sudo dpkg -i /tmp/firefox.deb || sudo apt-get -f install; \
+		sudo rm -rf /tmp/firefox.deb; \
+	else \
+		echo "Not installing Firefox either because this is a CI."; \
+	fi
+
+install_js_dependencies: ## Install dependencies for JS code.
+	curl -sL https://deb.nodesource.com/setup_11.x | sudo -E bash -
+	sudo apt-get install -y nodejs
+	npm install
```

Comment:
This is currently failing for me in the vagrant devstack with a
```
npm ERR! enoent ENOENT: no such file or directory, open '/home/vagrant/opencraft/node_modules/nyc/node_modules/handlebars/node_modules/source-map/package.json.3468720659'
```

Any ideas?

Reply:
@viadanna This might just be npm. Try removing `package-lock.json` and `node_modules` and try again.

Reply:
@viadanna Also, try it again. We are not actually using `nyc` so I deleted it from `package.json`

Reply:
Ok, could get it installed but now I'm seeing:
```
found 1 high severity vulnerability
  run `npm audit fix` to fix them, or `npm audit` for details
```
Can you update the necessary packages?

Reply:
@viadanna Done